src/lib/core/components/Form/types/array.ts

```diff
@@ -16,13 +16,39 @@ import {
     ValidatorsMap,
 } from './';
 
-export type ArrayInputProps = InputProps<FieldArrayValue, ArraySpec>;
-export type ArrayIndependentInputProps = IndependentInputProps<FieldArrayValue, ArraySpec>;
-export type ArrayLayoutProps = LayoutProps<FieldArrayValue, ArraySpec>;
+export type ArrayInputProps<InputComponentProps extends Record<string, any> = {}> = InputProps<
+    FieldArrayValue,
+    ArraySpec<undefined, InputComponentProps>
+>;
+export type ArrayIndependentInputProps<
+    InputComponentProps extends Record<string, any> = {},
+    LayoutComponentProps extends Record<string, any> = {},
+> = IndependentInputProps<
+    FieldArrayValue,
+    ArraySpec<undefined, InputComponentProps, LayoutComponentProps>
+>;
+
+export type ArrayLayoutProps<LayoutComponentProps extends Record<string, any> = {}> = LayoutProps<
+    FieldArrayValue,
+    ArraySpec<undefined, any, LayoutComponentProps>
```

Comment:
InputComponentProps here will be undefined each time https://github.com/gravity-ui/dynamic-forms/pull/164/files#diff-303ba96de5f2391fcd13827b52f252634d812d40d54a7d01abb517cb5595cd4dR161